haystack/preview/components/generators/openai/gpt35.py

```diff
@@ -130,70 +130,63 @@ def from_dict(cls, data: Dict[str, Any]) -> "GPT35Generator":
-        for prompt in prompts:
-            message = _ChatMessage(content=prompt, role="user")
-            if self.system_prompt:
-                chats.append([_ChatMessage(content=self.system_prompt, role="system"), message])
-            else:
-                chats.append([message])
-
-        all_replies, all_metadata = [], []
-        for chat in chats:
-            completion = openai.ChatCompletion.create(
-                model=self.model_name,
-                api_key=self.api_key,
-                messages=[asdict(message) for message in chat],
-                stream=self.streaming_callback is not None,
-                **self.model_parameters,
-            )
+        message = _ChatMessage(content=prompt, role="user")
+        if self.system_prompt:
+            chat = [_ChatMessage(content=self.system_prompt, role="system"), message]
+        else:
+            chat = [message]
 
-            replies: List[str]
-            metadata: List[Dict[str, Any]]
-            if self.streaming_callback:
-                replies_dict = {}
-                metadata_dict: Dict[str, Dict[str, Any]] = {}
-                for chunk in completion:
-                    chunk = self.streaming_callback(chunk)
-                    for choice in chunk.choices:
-                        if choice.index not in replies_dict:
-                            replies_dict[choice.index] = ""
-                            metadata_dict[choice.index] = {}
-
-                        if hasattr(choice.delta, "content"):
-                            replies_dict[choice.index] += choice.delta.content
-                        metadata_dict[choice.index] = {
-                            "model": chunk.model,
-                            "index": choice.index,
-                            "finish_reason": choice.finish_reason,
-                        }
-                all_replies.append(list(replies_dict.values()))
-                all_metadata.append(list(metadata_dict.values()))
-                self._check_truncated_answers(list(metadata_dict.values()))
+        completion = openai.ChatCompletion.create(
+            model=self.model_name,
+            api_key=self.api_key,
+            messages=[asdict(message) for message in chat],
+            stream=self.streaming_callback is not None,
+            **self.model_parameters,
+        )
 
-            else:
-                metadata = [
-                    {
-                        "model": completion.model,
+        replies: List[str]
+        metadata: List[Dict[str, Any]]
+        if self.streaming_callback:
+            replies_dict = {}
+            metadata_dict: Dict[str, Dict[str, Any]] = {}
+            for chunk in completion:
+                chunk = self.streaming_callback(chunk)
+                for choice in chunk.choices:
+                    if choice.index not in replies_dict:
+                        replies_dict[choice.index] = ""
+                        metadata_dict[choice.index] = {}
```

Comment:
nit: you can spare these 3 lines using a defaultdict for both `repliease_dict` and `metadata_dict`